src/xvectorbin/nnet3-xvector-compute.cc

```diff
@@ -0,0 +1,112 @@
+// nnet3bin/nnet3-compute.cc
+
+// Copyright 2012-2015   Johns Hopkins University (author: Daniel Povey)
+//                2016   David Snyder
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "base/timer.h"
+#include "nnet3/nnet-utils.h"
+#include "xvector/nnet-xvector-compute.h"
+
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace kaldi::nnet3;
+    typedef kaldi::int32 int32;
+    typedef kaldi::int64 int64;
+
+    const char *usage =
+      "Propagate the features through the network and write the output\n"
+      "xvectors.  The xvector system assumes that for one input feature\n"
+      "matrix, regardless of the number of rows, there is only one output\n"
+      "xvector.  If the input has more rows than the network context, the\n"
```

Comment:
you have to be a bit careful how you define context here.
The left-context and right-context computed in the normal way are the *minimum* amount of context needed to compute an output.  However, what you want here is the *maximum* amount of context needed-- currently there is no general-purpose way to compute this.
It may be easiest in the short term to have the context supplied via the command line.